platform/security/core/security-core-impl/src/main/java/ddf/security/ws/policy/impl/PolicyWSDLGetInterceptor.java

```diff
@@ -102,6 +123,52 @@ public void handleMessage(Message message) throws Fault {
         }
     }
 
+    private Document configureAddress(Document doc) {
+        Element de = doc.getDocumentElement();
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        String xPathString = String
+                .format("//@*[starts-with(.,\"%s\")] | //text()[starts-with(.,\"%s\")]",
+                        DEFAULT_ADDRESS, DEFAULT_ADDRESS);
+        NodeList nodes;
+        try {
+            nodes = (NodeList) xPath.evaluate(xPathString, de, XPathConstants.NODESET);
+        } catch (XPathExpressionException e) {
+            LOGGER.error("Could not evaulate the xPath expression: {}", xPathString, e);
+            return doc;
+        }
+        for (int i = 0; i < nodes.getLength(); i++) {
+            Node node = nodes.item(i);
+            node.setNodeValue(node.getNodeValue().replaceAll(DEFAULT_ADDRESS, getAddress()));
+        }
+        return de.getOwnerDocument();
+    }
+
+    private String getAddress() {
+        return String.format("%s%s:%s", protocol, host, port);
+    }
+
+    @Override
+    public synchronized void configurationUpdateCallback(Map<String, String> properties) {
+        LOGGER.debug("Updating configuration from: {}", getAddress());
+        if (properties != null && !properties.isEmpty()) {
+            String protocol = properties.get(ConfigurationManager.PROTOCOL);
+            if (protocol != null && !protocol.isEmpty()) {
```

Comment:
:+1: 

Hell, if you wanted one-liners, you could combine `StringUtils` with a ternary operator.